Event/MailSubscriber.php

```diff
@@ -0,0 +1,217 @@
+use Sulu\Bundle\FormBundle\Entity\FormTranslation;
+use Sulu\Bundle\FormBundle\Mail\HelperInterface as MailHelperInterface;
+use Sulu\Bundle\MediaBundle\Media\Manager\MediaManager;
+use Sulu\Bundle\MediaBundle\Media\Storage\StorageInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Templating\EngineInterface;
+
+class MailSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var MailHelperInterface
+     */
+    protected $mailHelper;
+
+    /**
+     * @var EngineInterface
+     */
+    protected $templating;
+
+    /**
+     * @var MediaManager
+     */
+    protected $mediaManager;
+
+    /**
+     * @var StorageInterface
+     */
+    protected $storageManager;
+
+    /**
+     * @var string
+     */
+    protected $notifyTemplate;
+    /**
+     * @var string
+     */
+    protected $customerTemplate;
+
+    /**
+     * @param MailHelperInterface $mailHelper
+     * @param EngineInterface $templating
+     * @param MediaManager $mediaManager
+     * @param StorageInterface $storageManager
+     * @param string $notifyTemplate
+     * @param string $customerTemplate
+     */
+    public function __construct(
+        MailHelperInterface $mailHelper,
+        EngineInterface $templating,
+        MediaManager $mediaManager,
+        StorageInterface $storageManager,
+        $notifyTemplate,
+        $customerTemplate
+    ) {
+        $this->mailHelper = $mailHelper;
+        $this->templating = $templating;
+        $this->mediaManager = $mediaManager;
+        $this->storageManager = $storageManager;
+        $this->notifyTemplate = $notifyTemplate;
+        $this->customerTemplate = $customerTemplate;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            DynFormSavedEvent::NAME => 'handleMailDelivery',
+        ];
+    }
+
+    /**
+     * Handles the mail delivery on saving a dynamic form.
+     *
+     * @param DynFormSavedEvent $event
+     */
+    public function handleMailDelivery(DynFormSavedEvent $event)
+    {
+        $dynamic = $event->getDynamic();
+        $form = $dynamic->form;
+        $formEntity = $form->serializeForLocale($dynamic->locale, $dynamic);
+        $translation = $form->getTranslation($dynamic->locale);
+
+        if (!$translation->getDeactivateCustomerMails() && !empty($dynamic->email)) {
+            $customerMail = $this->templating->render($this->customerTemplate, ['formEntity' => $formEntity]);
+            $this->mailHelper->sendMail(
+                $translation->getSubject(),
+                $customerMail,
+                $dynamic->email,
+                $this->getFromAddress($translation),
+                true
+            );
+        }
+
+        if (!$translation->getDeactivateNotifyMails()) {
+            $allReceivers = [
+                MailHelperInterface::MAIL_RECEIVER_TO => [],
+                MailHelperInterface::MAIL_RECEIVER_CC => [],
```

Comment:
we should move this to the mail helper service and add a getReceiverTypes function.